### PR TITLE
Whats new

### PR DESCRIPTION
**Pull request by przybylski**
I finally got time to start merge whats new to main repo as mentioned in https://github.com/nextcloud/android/issues/50.

Re-order:
- [x] Manage all your files
- [x] ~~share~~
- [x] instant upload
- [x] Multi account
- [x] remove "Welcome to"
- [x] check font size: 50-100% larger
- [x] All Feature images need to have the same size
- [x] Test what's new for fresh install
- [x] Test what's new while upgrading

Bugs: 
- [x] features shown twice - Reopenend, to be checked by @AndyScherzinger --> cherry picked @tobiasKaminsky's commit from beta: https://github.com/nextcloud/android/pull/84/commits/b472fb3ecc120635466cd21646a9a782ac4b42b4

### CODE REVIEW

File: res/drawable/indicator_dot_not_selected.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ownCloud Android client application
+
+  Copyright (C) 2015 Bartosz Przybylski
+  Copyright (C) 2015 ownCloud Inc.
+
```
Comment (tobiasKaminsky) [inline res/drawable/indicator_dot_not_selected.xml]: also "nextcloud"?

File: res/drawable/indicator_dot_not_selected.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:bottom="5dp" android:top="5dp" android:right="5dp" android:left="5dp">
+
+        <shape
+            android:shape="oval">
+            <solid android:color="@color/owncloud_blue_dark_transparent"/>
```
Comment (tobiasKaminsky) [inline res/drawable/indicator_dot_not_selected.xml]: is this adapted to the new nextcloud colour?

File: res/drawable/indicator_dot_selected.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:bottom="5dp" android:top="5dp" android:right="5dp" android:left="5dp">
+        <shape
+            android:shape="oval">
+            <solid android:color="@color/owncloud_blue_accent"/>
```
Comment (tobiasKaminsky) [inline res/drawable/indicator_dot_selected.xml]: sam?

File: res/drawable/round_button.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="oval">
+    <solid android:color="@color/owncloud_blue_accent"/>
```
Comment (tobiasKaminsky) [inline res/drawable/round_button.xml]: same?

File: res/values/strings.xml [hunk 5]
```diff
@@ -459,4 +459,23 @@
     <string name="permission_storage_access">Additional permissions required to upload &amp; download files.</string>
     <string name="local_file_not_found_toast">The file was not found in the local file system</string>
 
+
+    <!-- What's new feature and texts to show -->
+    <string name="welcome_to_oc_title">Welcome to nextCloud!</string>
```
Comment (tobiasKaminsky) [inline res/values/strings.xml]: nextCloud -> Nextcloud

**Comment by tobiasKaminsky** [comment]
- the feature images must be adapted to the new nextcloud design @jancborchardt 
- do we want to remove beta support, since in the nearer future there will no beta, but only release candidates? @przybylski @AndyScherzinger

**Comment by przybylski** [comment]
@tobiasKaminsky I can remove or leave it. But if beta releases are planned then I would prefer to leave it as it is.

File: res/drawable/indicator_dot_not_selected.xml (hunk 2 above)
Comment (przybylski) [inline res/drawable/indicator_dot_not_selected.xml]: No, I'm just using colors already defined in resources

**Comment by AndyScherzinger** [comment]
I would keep the beta support for now since we still have a beta. So removing the beta related code could be done whenever we stop doing betas.

File: res/drawable/indicator_dot_not_selected.xml (hunk 2 above)
Comment (AndyScherzinger) [inline res/drawable/indicator_dot_not_selected.xml]: I can check for the colors, but should have adapted most values, so working with the references should work perfectly fine, but I'll do a check just to make sure I set all values :)

**Comment by AndyScherzinger** [comment]
* [ ] Manage all your files
* [ ] share
* [ ] instant upload
* [ ] Multi account

* [ ] Welcome to (remove)
* [ ] font sizechecken 50-100% larger

* [ ] All Feature images need to have the same size

Bugs: 

* [ ] features shown twice

**Comment by AndyScherzinger** [comment]
ping @eppfel :grin: 

device-2016-09-18-182923
device-2016-09-18-182829
device-2016-09-18-182837
device-2016-09-18-182915

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can you push your fix for "showing twice". It doesn't seem to be on the branch. :rocket:

**Comment by eppfel** [comment]
So, here are my drafts. Tried only with blue background so far: I feel it fits better for intro screens.
They are not finished (especially multiuser). Wanted to get some opinion before I put more work in it. @jancborchardt ? ..as I hijacked this. 😄 
I'm happy to do more for future releases, as long as you give an early heads-up.

multiaccount

files_list
instantupload
sharing

**Comment by tobiasKaminsky** [comment]
@eppfel I like :heart: them! :+1:

**Comment by AndyScherzinger** [comment]
Uuuuuh nice @eppfel can you provide the image without the background color? :+1:

**Comment by AndyScherzinger** [comment]
@nextcloud/designers and @eppfel here are the screenshots of the actual implementation incorporating the new feature graphics. Go/No-Go? 

One thing though @eppfel the Android app'S facebook/twitter etc. share is at the moment a download-and-share-file sharing not a link to a Nextcloud instance. Other than that: awesome job! :rocket: 

device-2016-09-27-002525

device-2016-09-27-002538

device-2016-09-27-002549

device-2016-09-27-002822

**Comment by te-online** [comment]
@eppfel Looks great! 😊  One question, though: Is it just me or is the image for ”multi account“ slightly off to the left? Or is there a reasoning behind this? 😅

**Comment by AndyScherzinger** [comment]
@te-online nice catch with the skip button, yes that will have to be removed for the last one. Will do. 

As for the multi account image, it is slightly off due to the fact that the images are horizontally centered and I guess due to the shadow is has a bit off. @eppfel can you add an adequate amount of margin to the left side if the image to balance it when centered :smiley:

**Comment by eppfel** [comment]
@te-online My intention was to hint on the menu being on the left side of the screen. Maybe the shadow is enough for that and the positioning looks odd. Will have to look into it. And again I feel this particular graphic is not finished...

**Comment by AndyScherzinger** [comment]
"skip" button is now hidden on the last feature element

device-2016-09-27-141423

**Comment by Espina2** [comment]
Hi everybody.

My thougts about ita like we should decide what style of "display" icons we should use, and waht color palete use to. For example that purple, green a black blue I didnt see it anywhere. 

We are in part of the design process that we should think a about what we do, and no only that what looks nice. 

We should start design like a brand and not have awesome mocks that whem we group them all, looks like separate things.

**Comment by eppfel** [comment]
@Espina2 I totally agree, that's why I call these drafts (the colors are not final). I am currently trying to start building a Design Guide nextcloud/nextcloud.com#140 Discussing things like this, should help us build it.

**Comment by Espina2** [comment]
I have to take a look. What im trying to say is that if you take a look at what I do for the website its very different that you make here. Im not saying that my version of icons is better than yours, only saying that we should decide where to go, and applying the same values and thougts in all the cases (apps, website, flyers).

**Comment by eppfel** [comment]
@Espina2 ..and I agree again. 😁 That's why we need a place, where we collect and define these guidelines.
I will have to take a closer look at your redesign of the website. I used current icons from nextcloud for consistency and adapted some visual cues known to Android users. As I'm not a visual designer, I am open to any help.

**Comment by Espina2** [comment]
@eppfel Nice that we agree. I think the problem its not having a Styleguide, nowadays we need brand guidelines, a brand its not a logo is a ecosystem, and all the values should be the same no mather the media we are using. 

I think you cant do a Styleguide without brand guidelines

Just to point and why the things are different I will put one example of Brand Guidelines and a Styleguide.

Brand Guidelines https://www.edp.pt/en/aedp/sobreaedp/marcaEDP/EDPDocs/EDPBrandGuide_2013_09_11_EN.pdf

Styleguide
https://www.lightningdesignsystem.com/

Let me think what you think about it. :)

**Comment by eppfel** [comment]
So, we just discussed it is better to have something for now than nothing. I will redesign mutli account anyhow. These should be refreshed every release anyway. Then we can streamline this to the new overall look.

What do you think @jancborchardt ?

**Comment by tobiasKaminsky** [comment]
@eppfel great to see a progress :+1:

**Comment by AndyScherzinger** [comment]
Just had the first Rhein Main Meetup

**Comment by jancborchardt** [comment]
@eppfel awesome, sorry for the late reply! I have only one main feedback: We should combine the »Manage your files« and »Share« screens. More than 3 screens is a bit much.

I would even make it the »A safe home for all your data« slogan from the website, and a shorter version of the subtitle from the website as subtitle: »Access, share & protect your files at home and in your enterprise«. Then a collage of some file type icons and the sharing icon or so. What do you think?

**Comment by eppfel** [comment]
@jancborchardt This is an excellent idea, I will get to it.

**Comment by eppfel** [comment]
So, the new Instant Upload is in for the beta #358 

>> Awesome work, now I have to add something to the whatsnew
> Looking forward to this too 😃

I just realized, I created an instant upload whatsnew already. I will revise this one to point out, that this now supports complete folders, not just the camera.

**Comment by AndyScherzinger** [comment]
@eppfel just FYI, the next RC/stable release will ship the SD card support :grin:

**Comment by AndyScherzinger** [comment]
@eppfel any update on the MultiAccount image? :smiley:

**Comment by eppfel** [comment]
@AndyScherzinger Sry, no. Is there a deadline or can this wait for our meetup next week?

**Comment by AndyScherzinger** [comment]
@eppfel next is absolutely fine but I probably won't make it since I have a GoLive in my day job project :(

**Comment by tobiasKaminsky** [comment]
Please check why the screens are launched twice.

**Comment by AndyScherzinger** [comment]
@eppfel any update on the MultiAccount image?

**Comment by AndyScherzinger** [comment]
@eppfel any news?
@tobiasKaminsky I vote to just merge, one image is slightly off, but it is like that since before 27th of Sept this can be fixed after the release or in a later RC. To get this thing solved it need to get merged to create business pain ;)

**Comment by eppfel** [comment]
Ah, yeah I just spent a bit on it, but the multiaccount is not ready. I like the proposal of @jancborchardt to combine files and sharing. I can upload something later tonight.
Regarding the auto upload: Is the camera - photos graphic relevant or Misleading?

**Comment by AndyScherzinger** [comment]
I think the instant upload is fine for now. When we roll out the Auto upload to more Android versions than we can change this, since it is then a auto media upload and not restricted to the camera. 
Something later tonight sounds perfectly fine to me, then I can do a final test and merge to master, so it can be shipped with 1.4.0 😃 
Will the images then be done for an initial release (done is better than perfect ;)

**Comment by eppfel** [comment]
@AndyScherzinger Can you have a look? All graphics should be of equal size and are ok to release. (Although screenshots would be nice for a final review). Can you remove the sharing step and change the text as propose by @jancborchardt ?

**Comment by AndyScherzinger** [comment]
@eppfel looks good and works like a charm 👍

**Comment by AndyScherzinger** [comment]
device-2016-11-23-211237
device-2016-11-23-211245
device-2016-11-23-211252

**Comment by eppfel** [comment]
mh, little bit boring, but yeah, good enough.
@AndyScherzinger Could we make the order: files -> multi account -> upload? I feel this is more logical order, because it goes from essential to special. And shouldn't we say "Auto Upload" now?

**Comment by AndyScherzinger** [comment]
Will change that, Auto upload is only activated for Android 7 so we should probably wait until we can activate it for Android 5 and 6 too. I am changing the text as proposed by Jan right now and will post the new screenshots in a couple of minutes

**Comment by AndyScherzinger** [comment]
@eppfel final check please :)

device-2016-11-23-213844
device-2016-11-23-213853
device-2016-11-23-213900

**Comment by AndyScherzinger** [comment]
@eppfel ?final comment?

**Comment by eppfel** [comment]
Margin of the subtitles maybe a little bigger. Otherwise : GO! 🚀

**Comment by AndyScherzinger** [comment]
device-2016-11-23-213844

**Comment by AndyScherzinger** [comment]
@eppfel like this? :)

device-2016-11-23-222556

**Comment by eppfel** [comment]
:blue_heart:

**Comment by AndyScherzinger** [comment]
So here are the final final screenshots, as per latest discussion and review of @eppfel 

device-2016-11-23-223058
device-2016-11-23-223108
device-2016-11-23-223116

**Comment by jancborchardt** [comment]
This is damn awesome, and looks very nice! :) Great work folks!

**Comment by AndyScherzinger** [comment]
Thanks @jancborchardt!

A joint effort of @przybylski, @tobiasKaminsky, @eppfel, @AndyScherzinger :)

**Comment by eppfel** [comment]
Works, great in the app, too! 👍

**Comment by jancborchardt** [comment]
The Android team rocks and we should have more blog posts about the cool enhancements. ;)

cc @jospoortvliet

**Comment by AndyScherzinger** [comment]
Yeah, yeah.... already planed to write the piece for 1.4.0 :) No worries there will be an email in Jos' inbox these days to blog about it right after the launch or some days later...

**Comment by enoch85** [comment]
Great work guys, just as a remark - in the translation I used "Automatic upload" instead of "Instant Upload" everywhere. 

Now when we rename it, why not just rename everything?

**Comment by AndyScherzinger** [comment]
Because auto upload is the new instant upload which is only activated for Android 7 with the first release. When we activate it for Android 6 and 5 it will then be renamed. Except for the old, legacy string for the setting Screen for Android for, that will keep the old terms to not raise expectations we can't fulfill.

**Pull request merged**